scripts/dice.coffee

```diff
@@ -40,18 +46,30 @@ module.exports = (robot) ->
       report fudgeRoll dice
     msg.reply answer
 
-report = (results) ->
+report = (modifier, results) ->
   if results?
     switch results.length
       when 0
         "I didn't roll any dice."
       when 1
-        "I rolled a #{results[0]}."
+        total = results[0]
+        "I rolled a #{total}."
+        if modifier
+          modified total, modifier
       else
         total = results.reduce (x, y) -> x + y
-        finalComma = if (results.length > 2) then "," else ""
-        last = results.pop()
-        "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        if results.length < 10
+          finalComma = if (results.length > 2) then "," else ""
+          last = results.pop()
+          "I rolled #{results.join(", ")}#{finalComma} and #{last}, making #{total}."
+        else
+          "I rolled a handful of dice, making #{total}."
```

Comment:
This if/else pair probably needs to be set to some variable than you append the modifier stuff to afterwards and then return.